PowerEditor/src/NppIO.cpp

```diff
@@ -1598,26 +1598,54 @@ bool Notepad_plus::fileSaveSpecific(const generic_string& fileNameToSave)
 	}
 }
 
-bool Notepad_plus::fileSaveAll()
+bool Notepad_plus::fileSaveAllConfirm()
 {
-	if (viewVisible(MAIN_VIEW))
+	bool confirmed = false;
+
+	if (NppParameters::getInstance().getNppGUI()._saveAllConfirm)
 	{
-		for (size_t i = 0; i < _mainDocTab.nbItem(); ++i)
+		int answer = _nativeLangSpeaker.messageBox("SaveAllConfirm",
+			_pPublicInterface->getHSelf(),
+			TEXT("Are you sure you want to save all documents??"),
+			TEXT("Are you sure?"),
+			MB_YESNO);
+
+		if (answer == IDYES)

```

Comment:
After block of `if (answer == IDYES)`
Add `if (answer == IDCANCEL)` to dis-activate `_saveAllConfirm`